include/CIL/Transformations.hpp

```diff
@@ -41,7 +41,7 @@ namespace CIL {
     void
     rotate(ImageInfo& img, int degrees,
            RotationKind rotation_kind = RotationKind::rotation_by_area_mapping);
-    void convertToGrayscale(ImageInfo& img);
+    void convertToGrayscale(ImageInfo& img, bool preserve_components = true);
```

Comment:
Can you please change `preserve_componenets` to `preserve_colortype`? `preserve_colortype` seems more intuitive.